packages/cursorless-engine/src/compositionRoot/createCursorlessEngine.ts

```diff
@@ -33,7 +36,12 @@ export function createCursorlessEngine(
 
   const testCaseRecorder = new TestCaseRecorder(hatTokenMap);
 
-  const actions = new Actions(snippets, rangeUpdater);
+  const scopeHandlerFactory = new ScopeHandlerFactoryImpl();
```

Comment:
@AndreasArvidsson notice how the `scopeHandlerFactory` is instantiated at the composition root, so there's no "deep drilling".  Anything that a scope handler needs to do its job just gets passed one level through the factory to the scope handler that needs it.  The decoupling actually results in shallower hierarchies, because the code that uses scope handlers isn't responsible for giving them their dependencies.  It also makes testing easier, because we could just pass in a fake scope handler factory to a modifier.  Tbh it would be a bit cleaner if we could just pass in a scope handler directly to a modifier, but the problem is that we might need a different scope handler per target, so it is useful for the modifier to be able to construct the scope handler on the fly